File: FixYaml.java
import java.io.*;
import java.nio.file.*;
import java.nio.charset.*;
import java.util.*;

public class FixYaml {
    public static void main(String[] args) throws Exception {
        File dir = new File("MistakenDeluxe-Core/src/main/resources/langs/es");
        for (File f : dir.listFiles()) {
            if (!f.getName().endsWith(".yml")) continue;
            
            byte[] bytes = Files.readAllBytes(f.toPath());
            
            // Remove 0x81
            ByteArrayOutputStream out = new ByteArrayOutputStream();
            for (byte b : bytes) {
                if ((b & 0xFF) != 0x81) {
                    out.write(b);
                }
            }
            byte[] newBytes = out.toByteArray();
            
            // Decode as Windows-1252 or fallback to UTF-8
            String text = new String(newBytes, StandardCharsets.UTF_8);
            
            // Replace strings
            text = text.replace("INFORMACI\uFFFD\"N", "INFORMACI\u00D3N");
            text = text.replace("INFORMACI\uFFFDN", "INFORMACI\u00D3N");
            text = text.replace("estad\uFFFDsticas", "estad\u00EDsticas");
            text = text.replace("Mec\u01ECnica", "Mec\u00E1nica");
            text = text.replace("cl\u01ECsico", "cl\u00E1sico");
            text = text.replace("f\u00EDsica", "f\u00EDsica");
            text = text.replace("fsica", "f\u00EDsica");
            text = text.replace("mgico", "m\u00E1gico");
            text = text.replace("F\u01ECcil", "F\u00E1cil");
            text = text.replace("persecucin", "persecuci\u00F3n");
            text = text.replace("Dif\uFFFDcil", "Dif\u00EDcil");
            text = text.replace("Mgica", "M\u00E1gica");
            text = text.replace("T\uFFFDnel", "T\u00FAnel");
            text = text.replace("\uFFFDltimo", "\u00DAltimo");
            text = text.replace("M\uFFFDsica", "M\u00FAsica");
            text = text.replace("Da\uFFFDo", "Da\u00F1o");

            Files.write(f.toPath(), text.getBytes(StandardCharsets.UTF_8));
            System.out.println("Cleaned " + f.getName());
        }
    }
}

File: fix_yaml.py
import os

def fix_file(filepath):
    with open(filepath, 'rb') as f:
        data = f.read()

    # 0x81 is the bad byte causing SnakeYAML to crash
    if b'\x81' in data:
        print(f"Found 0x81 in {filepath}, removing it.")
        data = data.replace(b'\x81', b'')
        
    # 'clǭsico' -> 'clásico' (C7 AD -> C3 A1)
    if b'\xc7\xad' in data:
        data = data.replace(b'\xc7\xad', b'\xc3\xa1')
        
    # 'fsica' -> 'física' (corrupted originally as fsica)
    # The previous script did these text replacements, so I will do text replacements after fixing bytes.
    try:
        text = data.decode('utf-8')
        text = text.replace('fsica', 'física')
        text = text.replace('mgico', 'mágico')
        text = text.replace('persecucin', 'persecución')
        text = text.replace('Fǭcil', 'Fácil')
        data = text.encode('utf-8')
    except Exception as e:
        print(f"Failed to decode {filepath}: {e}")

    with open(filepath, 'wb') as f:
        f.write(data)

base_path = r"c:\Users\L900m\OneDrive\Desktop\MistakenDeluxe\MistakenDeluxe\MistakenDeluxe-Core\src\main\resources\langs\es"
for f in ["messages.yml", "asesinos_info.yml", "supervivientes_info.yml"]:
    fix_file(os.path.join(base_path, f))

print("Done fixing files!")
